Feature/advance docs (#35)
infinum/eightshift-docs: docs/guides/di-container.md

title: Dependency injection container
---

Dependency injection is a way of implementing inversion of control design pattern. It's used to handle the dependencies between multiple classes without the direct instantiation of one class in another - which causes tight coupling in the code and makes it less testable and harder to maintain.

> In general, if you have something like this in your class `new SuperClass()` that is bad.

Reviewer: ```suggestion
In general, if you have something like this in your class: `new SuperClass()` it's bad.
```

> In general, if you have something like this in your class `new SuperClass()` that is bad.

Reviewer 2: I'd split it and reword it a bit:

---

In general, if you instantiate another class like this inside your class

```php
new SuperClass()
```

that should be avoided because it tightly couples two classes.

---

Even though it's mentioned in the paragraph above.
